src/views/Studies/components/PageContent/Guid/CavaticaGuidModal.tsx

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useState } from 'react';
+import intl from 'react-intl-universal';
+import { PlusOutlined } from '@ant-design/icons';
+import {
+  CAVATICA_TYPE,
+  ICavaticaTreeNode,
+} from '@ferlab/ui/core/components/Widgets/Cavatica/CavaticaAnalyzeModal';
+import { ICavaticaProjects } from '@ferlab/ui/core/components/Widgets/Cavatica/type';
+import { Button, Divider, Modal, ModalFuncProps, Space, TreeSelect, Typography } from 'antd';
+import { LegacyDataNode } from 'rc-tree-select/lib/TreeSelect';
+
+import styles from './index.module.css';
+
+export interface ICavaticaAnalyseModal extends ModalFuncProps {
+  handleCreateProjectClick?: () => void;
+  handleFilesAndFolders: (parentId: string, isProject: boolean) => any;
+  handleSubmit: (value: ICavaticaTreeNode) => void;
+  onClose: () => void;
+  open: boolean;
+  projects: ICavaticaProjects;
+}
+
+const CavaticaGuidModal = ({
+  handleCreateProjectClick,
+  handleFilesAndFolders,
+  handleSubmit,
+  onClose,
+  open,
+  projects,
+}: ICavaticaAnalyseModal): JSX.Element => {
+  const [selectedTreeNode, setSelectedTreeNode] = useState<ICavaticaTreeNode | undefined>();
+  const [localProjectTree, setLocalProjectTree] = useState<ICavaticaTreeNode[]>([]);
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const NewProjectButton = () => (
```

Comment:
Could be defined outside the component with a Prop (for click) to avoid redefenition on everyRender